src/app/configurator/configurator.component.html

```diff
@@ -1 +1,76 @@
-<h1>Configurator works!</h1>
+<div class="dialog-window">
+
+  <h1 mat-dialog-title>{{configurator | uppercase }} - ID {{productId}}</h1>
+
+  <form [formGroup]="configuratorForm" (ngSubmit)="submit()">
+    <formly-form [form]="configuratorForm">
+
+      <mat-dialog-content class="mat-typography">
+        <div class="wrapper">
+
+          <div class="column">
+
+            <div>
+              <h3>Start date: </h3>
+              <mat-form-field appearance="fill">
+                <mat-label>Select Date</mat-label>
+                <input formControlName="date" matInput [min]="minDate" matInput [matDatepicker]="datepicker">
+                <mat-datepicker-toggle matSuffix [for]="datepicker"></mat-datepicker-toggle>
+                <mat-datepicker #datepicker>
+                  <mat-datepicker-actions>
+                    <button mat-button color="warn" matDatepickerCancel>Cancel</button>
+                    <button mat-raised-button color="primary" matDatepickerApply>Apply</button>
+                  </mat-datepicker-actions>
+                </mat-datepicker>
+              </mat-form-field>
+            </div>
+
+            <div *ngIf="configurator !== 'trial'">
+              <h3>Subscription period:</h3>
+              <mat-form-field appearance="fill">
+                <mat-label>Select Duration</mat-label>
+                <input formControlName="duration" matInput type="number" min="1" max="24">
+                <mat-hint>months</mat-hint>
+              </mat-form-field>
+            </div>
+
+            <div *ngIf="configurator === 'trial'">
+              <h3>Subscription period:</h3>
+              <mat-form-field appearance="fill">
+                <mat-label>Select Duration</mat-label>
+                <input formControlName="duration" matInput type="number" min="6" max="6">
+                <mat-hint>6 months</mat-hint>
+              </mat-form-field>
+            </div>
+
+
+            <div class="support" *ngIf="configurator !== 'free' && configurator !== 'trial'">
+              <h3 class="txt-support">Support Service: </h3>
+              <mat-checkbox formControlName="support">
+                Support
+              </mat-checkbox>
+            </div>
```

Comment:
The idea was to distinguish the different form with the FormlyFieldConfig. Please look up at how that works.

Reply:
unfortunately this is not done :)